src/components/network/network-panel.js

```diff
@@ -129,14 +142,44 @@ genotet.NetworkPanel.prototype.setNodeInfo_ = function(node, container) {
  * @private
  */
 genotet.NetworkPanel.prototype.setEdgeInfo_ = function(edge, container) {
+  var sourceLabel, targetLabel;
+  var sourceId, targetId;
+  if (typeof edge.source == 'object') {
```

Comment:
Can't we change edge to genotet.networkEdge and avoid this type checking? That is to make everything passed here have the same structure as genotet.networkEdge.